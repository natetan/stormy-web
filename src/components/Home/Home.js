import Geocode from 'react-geocode';
import React, { useState, useEffect } from 'react';

import { getWeatherInfo } from '../../services/weatherService';
import { tryRequire } from '../../lib/utils';
import MastHead from '../MastHead/MastHead';

import './Home.scss';

const Home = () => {
  const [position, setPosition] = useState({});
  const [location, setLocation] = useState('');
  const [desc, setDesc] = useState('');
  const [error, setError] = useState(null);
  const [search, setSearch] = useState('');

  let apiKey;
  try {
    let auth = require('../../auth.json');
    apiKey = auth.google_api_key;
  } catch (e) {
    console.log(e);
    apiKey = process.env.google_api_key;
  }
  Geocode.setApiKey(apiKey);

  const onChange = ({ coords }) => {
    setPosition({
      latitude: coords.latitude,
      longitude: coords.longitude
    });
  };

  const onError = error => {
    setError(error.message);
  }

  useEffect(() => {
    const gl = navigator.geolocation;
    if (!gl) {
      setError('Geolocation is not supported :(');
      return;
    }
    let watcher = gl.watchPosition(onChange, onError);
    return () => gl.clearWatch(watcher);
  }, [])

  if (!location && (position.latitude && position.longitude)) {
    Geocode.fromLatLng(position.latitude, position.longitude).then(response => {
      let city = response.results[0].address_components[3].long_name;
      getWeatherInfo(position.latitude, position.longitude).then(res => {
        console.log(res);
        let temp = Math.round(res.currently.temperature);
        setDesc(res.currently.summary);
        setLocation(`${city} | ${temp} F°`);
      });
    });
  }

  const handleInput = query => {
    setSearch(query);
  }

  const handleEnter = event => {
    if (event.key === 'Enter' && search.length) {
      handleSearch();
    }
  }

  const handleSearch = () => {
    Geocode.fromAddress(search).then(response => {
      if (response) {
        let results = response.results;
        if (results.length) {
          let lat = results[0].geometry.location.lat;
          let long = results[0].geometry.location.lng;
          let city = results[0].address_components[0].long_name;
          console.log(results)
          getWeatherInfo(lat, long).then(res => {
            console.log(res);
            let temp = Math.round(res.currently.temperature);
            setDesc(res.currently.summary);
            setLocation(`${city} | ${temp} F°`);
          });
        }
      }
    });
  }

  return (
    <div className='home-page'>
      <div className='cover-container d-flex h-100 p-3 mx-auto flex-column'>
        <MastHead
          title={`${location ? location : 'Stormy'}`}
          subtitle={`${desc ? desc : 'Demo weather app'}`}
          handleInput={handleInput}
          handleEnter={handleEnter}
          onClick={handleSearch}
        />
      </div>
    </div>
  )
}

export default Home;