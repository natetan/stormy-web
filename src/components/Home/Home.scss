.home-page {
  .cover-container {

    .mb-auto,
    .my-auto {
      margin-bottom: 5rem !important;
    }
  }
}
