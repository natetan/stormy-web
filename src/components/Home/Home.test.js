import React from "react";
import { shallow, mount } from 'enzyme';
import Home from './Home';

const weatherService = require('../../services/weatherService');
jest.mock('../../services/weatherService');

describe('Home component', () => {

  afterEach(() => {
    weatherService.getWeatherInfo.mockRestore();
  })

  it('renders correctly', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
})