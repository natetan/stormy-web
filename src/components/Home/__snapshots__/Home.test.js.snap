// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `"<div class=\\"home-page\\"><div class=\\"cover-container d-flex h-100 p-3 mx-auto flex-column\\"><header class=\\"masthead\\"><div class=\\"container d-flex h-100 align-items-center\\"><div class=\\"mx-auto text-center\\"><h1 class=\\"mx-auto my-0 text-uppercase\\">Stormy</h1><h2 class=\\"text-white-50 mx-auto mt-2 mb-5\\">Demo weather app</h2><div class=\\"search-bar\\"><input type=\\"text\\" placeholder=\\"Type in a location of some sort\\" class=\\"mr-sm-2 form-control\\"/></div><button type=\\"button\\" class=\\"btn btn-primary btn btn-primary\\">Search</button></div></div></header></div></div>"`;
